examples/react-native/.babelrc

```diff
@@ -1,3 +1,3 @@
 {
-  "presets": ["env", "react-native"]
+  "presets": ["@babel/preset-env", "module:metro-react-native-babel-preset"]
 }
```

Comment:
@SimenB @loganfsmyth 

Documentation for future debugging.

RN tests are currently broken and I've been unable to resolve them. The `examples/react-native` package includes the Env and Metro presets (which includes class properties, etc). When tests are ran, RN fails with the following:

```
 FAIL  examples/react-native/__tests__/intro.test.js
  ● Test suite failed to run

    SyntaxError: /Users/Miles/Sites/jest/examples/react-native/node_modules/react-native/jest/mockComponent.js: Support for the experimental syntax 'classProperties' isn't currently enabled (20:24):

      18 | 
      19 |   const Component = class extends SuperClass {
    > 20 |     static displayName = 'Component';
         |                        ^
      21 | 
      22 |     render() {
      23 |       const name =

    Add @babel/plugin-proposal-class-properties (https://git.io/vb4SL) to the 'plugins' section of your Babel config to enable transformation.

      at _class.raise (../../node_modules/@babel/parser/lib/index.js:3938:15)
```

Since this component is within node_modules, and Babel no longer transpiles node_modules, the class properties are breaking the test. Furthermore, since `/Users/Miles/Sites/jest/examples/react-native/node_modules/react-native/package.json` exists, Babel would exit early since it "found" a config file, instead of moving up to `/Users/Miles/Sites/jest/examples/react-native/.babelrc`, in which the Metro preset exists (which includes class properties).

I tried adding the class properties plugin to the root `babel.config.js` file in Jest, which resolved the RN issue, BUT actually broke a ton of other Jest tests as the compiled output is no longer what Jest expects.

At the moment, I don't believe this issue can easily be resolved with the knowledge I have.